x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/FileOperatorUsersStore.java

```diff
@@ -129,49 +143,81 @@ static final class Group {
         private static final Set<String> SINGLETON_REALM_TYPES = Set.of(
             FileRealmSettings.TYPE,
             NativeRealmSettings.TYPE,
-            ReservedRealm.TYPE
+            ReservedRealm.TYPE,
+            ServiceAccountSettings.REALM_TYPE
         );
 
         private final Set<String> usernames;
         private final String realmName;
         private final String realmType;
+        private final String tokenSource;
+        private final Set<String> tokenNames;
         private final Authentication.AuthenticationType authenticationType;
 
         Group(Set<String> usernames) {
             this(usernames, null);
         }
 
         Group(Set<String> usernames, @Nullable String realmName) {
-            this(usernames, realmName, null, null);
+            this(usernames, realmName, null, null, null, null);
         }
 
-        Group(Set<String> usernames, @Nullable String realmName, @Nullable String realmType, @Nullable String authenticationType) {
+        Group(
+            Set<String> usernames,
+            @Nullable String realmName,
+            @Nullable String realmType,
+            @Nullable String authenticationType,
+            @Nullable String tokenSource,
+            @Nullable Set<String> tokenNames
+        ) {
             this.usernames = usernames;
             this.realmName = realmName;
             this.realmType = realmType == null ? FileRealmSettings.TYPE : realmType;
             this.authenticationType = authenticationType == null
                 ? Authentication.AuthenticationType.REALM
                 : Authentication.AuthenticationType.valueOf(authenticationType.toUpperCase(Locale.ROOT));
+            this.tokenSource = tokenSource;
```

Comment:
I chose to model it as a string and the validate that string to get a meaningful error message. 
as-is the error message is "[token_source] must be one of the following values [index,file]"
..but if I modeled it as the enum the error message about failing to parse and no enum constant. It might be possible to move the validation up a couple lines or other ways to change the existing code such we can have both meaningful error messages and modeled as an enum but it feel like a decent amount of change for little benefit in this very small usage.